### PR TITLE
[IRRC-108 & 116] Changes to NewTeacher and EditTeacher form fields, added TeachingExperience to teacher model

### PR DESCRIPTION
**Pull request by nondescryptid**
# Frontend changes to NewTeacher and EditTeacher form fields

[IRRC-108](https://bootcamp-irrc.atlassian.net/browse/IRRC-108)

## Description
This PR implements changes for frontend by changing form fields in both NewTeacher and EditTeacher. Also has backend changes that allow you to create a teacher with TeachingExperience 

For both NewTeacher and EditTeacher: 
- English Proficiency field is no longer displayed
- Second Language Proficiency is no longer displayed
- Teaching Experience is displayed as a dropdown with "Yes" and "No" options 
- "Language 1" replaces "Native Language"
- "Language 2" replaces "Second Language"

NewTeacher only: 
- Added text above form fields to inform users that fields with * are required. While we have our own validation check that disables the Create Student button when not all required fields are filled in, I realise that users don't really have an indication of which fields are required. 

In NewTeacher, EnglishProficiency is no longer displayed, and is instead given a default value of "Intermediate" when new teachers are created (so the POST request won't fail because of constraints regarding English Proficiency). This is a workaround to avoid having to change our teacher model / changing database constraints. 

TeachingExperience is also displayed in EditTeacher and the teacher's ProfileCard on their profile 

## Test Steps

1. Create a new Teacher with the form (select TeachingExperience with the dropdown) with **2 languages** (this is because we can't test EditTeacher without a teacher with 2 languages at the moment). Also check if the asterisks + instructions that "fields with * are required" are there. 
2. Go to that teacher's profile and see if TeachingExperience is reflected on the profile card
3. Try to edit that teacher and change the value of TeachingExperience

### CODE REVIEW

File: backend/migrations/20201205141817_teachers.js [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
-        "Intermediate"
-      ])
-    table.text('Notes')
-    table.integer('StatusID').references('StatusID').inTable('statuses').onDelete("CASCADE")
-    table.timestamps(true, true)
-  })
-}
+      "Intermediate",
+    ]);
+    table.enu("TeachingExperience", ["Yes", "No"]);
```
Comment (thchia) [inline backend/migrations/20201205141817_teachers.js]: This should be in a new migration file.

**Review by thchia — CHANGES_REQUESTED** [review]
Untested, one comment on the migration. Whenever we amend the database schema (what columns there are or the column types etc.), we should create a _new_ migration file. This lets us undo that specific schema change if we need to.

File: backend/migrations/20201205141817_teachers.js (hunk 1 above)
Comment (xuanjane) [reply backend/migrations/20201205141817_teachers.js]: can create new migration file with command in backend: `npm run db:migrate:make add_teaching_experience`
where `add_teaching_experience` is the filename. You can use another name if you like as well

File: backend/src/models/teacher.js [hunk 2]
```diff
@@ -1,82 +1,83 @@
...
+    };
   }
 
-  static get jsonSchema () {
+  static get jsonSchema() {
     return {
-      type: 'object',
-      required: ['PhoneNumber', 'FullName', 'StatusID'],
+      type: "object",
+      required: ["PhoneNumber", "FullName", "StatusID"],
```
Comment (xuanjane) [inline backend/src/models/teacher.js]: I'm not sure if we should make `TeachingExperience` a `required` input as well? (if we are using it as part of the matching logic in future)

If we decide to make it a required input, then we will also need it to `.defaultTo("No")` in the migrations file 

```js
exports.up = function(knex, Promise) {
    return knex.schema.table(tableTeachers, (table) => {
        table.enu('TeachingExperience', ["Yes", "No"]).notNullable().defaultTo("No");
    });
};
```

Comment (nondescryptid) [reply backend/src/models/teacher.js]: Yup, we should make it a required input! thanks for bringing that up :D i've never made a migration file before.. haha

**Comment by nondescryptid** [comment]
@thchia @xuanjane Hi! Thank you so much for your comments. I've made a separate migration file to add TeachingExperience as a required input. For testing, you should be able to create a new teacher with TeachingExperience through the UI and also view their teaching experience on their individual profile cards!

File: backend/src/models/teacher.js (hunk 2 above)
Comment (xuanjane) [reply backend/src/models/teacher.js]: Haha great job Tomoe! Saw the `.notNullable().defaultTo("No")` in the migrations file :)
We can then also update the Teachers model to make TeachingExperience a required below (see below):

```suggestion
  static get jsonSchema() {
    return {
      type: "object",
      required: ["PhoneNumber", "FullName", "StatusID", "TeachingExperience"],
```

**Review by xuanjane — APPROVED** [review]
The edit button (to edit teacher/student details) is not working, but the bug is not from your PR.
So I am going to approve as my suggested change to your PR is a minor one :)

**Review by thchia — APPROVED** [review]
Marking as approved since my requested changes have been made.

**Pull request merged**